api/src/certification/enrolment/infrastructure/repositories/candidate-repository.js

```diff
@@ -20,7 +21,65 @@ const findBySessionId = async function ({ sessionId }) {
   return results.map(_toDomain);
 };
 
-export { findBySessionId };
+/**
+ * @function
+ * @param {Object} params
+ * @param {number} params.certificationCandidateId
+ *
+ * @return {Candidate}
+ */
+const get = async function ({ certificationCandidateId }) {
+  const certificationCandidate = await knex('certification-candidates')
+    .where({
+      id: certificationCandidateId,
+    })
+    .first();
+
+  return _toDomain(certificationCandidate);
+};
+
+/**
+ * @function
+ * @param {Object} certificationCandidate
+ *
+ * @return {Candidate}
+ * @throws {CertificationCandidateNotFoundError} Certification candidate not found
+ */
+const update = async function (certificationCandidate) {
+  const [updatedCertificationCandidate] = await knex('certification-candidates')
+    .where({
+      id: certificationCandidate.id,
+    })
+    .update({ ...certificationCandidate })
+    .returning('*');
+
+  if (!updatedCertificationCandidate) {
+    throw new CertificationCandidateNotFoundError();
+  }
+
+  return _toDomain(updatedCertificationCandidate);
+};
+
+/**
+ * @function
+ * @param {Object} params
+ * @param {number} params.certificationCandidateId
+ * @param {number} params.userId
+ *
+ * @return {Boolean} Returns true if candidate is found or false otherwise
+ */
+const isUserCertificationCandidate = async function ({ certificationCandidateId, userId }) {
+  const certificationCandidate = await knex('certification-candidates')
+    .where({
+      id: certificationCandidateId,
+      userId,
+    })
+    .first();
```

Comment:
Réduire légèremment le transfert de données sur le réseau

```suggestion
  const certificationCandidate = await knex.select(1).from('certification-candidates')
    .where({
      id: certificationCandidateId,
      userId,
    })
    .first();
```